vendor/github.com/operator-framework/api/pkg/operators/v1alpha1/installplan_types.go

```diff
@@ -297,7 +297,7 @@ func (b *BundleLookup) SetCondition(cond BundleLookupCondition) BundleLookupCond
 		if existing.Type != cond.Type {
 			continue
 		}
-		if existing.Status == cond.Status {
+		if existing.Status == cond.Status && existing.Reason == cond.Reason {
```

Comment:
Should we open up a PR in `api` for this?

Reply:
Good call - I will do that.